drivers/pwm/Kconfig.nxp_flexio

```diff
@@ -0,0 +1,11 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config PWM_NXP_FLEXIO
+	bool "NXP Flexio PWM driver"
+	default n
```

Comment:
Why is this `default n`? Typically drivers are `default y`, so they are enabled whenever the relevant API class is enabled and a device with the appropriate compatible is enabled (which will set the `DT_HAS_xxx_ENABLED` Kconfig)

Reply:
@danieldegrasse 
Twister build target for board twr_ke18f was failing for me, if I enabled flexio pwm driver by default 
west build -p always -b twr_ke18f  tests/drivers/build_all/pwm/
 
**Reason is**: KE1x family SoCs does have PWM support but doesn't have PINOUTD or PINOUTE registers, but these registers are required by S32 and MCX and i.Mx SoCs. 
So driver APIs that I added under FSL_FEATURE_FLEXIO_HAS_PIN_REGISTER are undefined for ke18f build target.

**Solution**: I can check for FSL_FEATURE_FLEXIO_HAS_PIN_REGISTER for Pin related API calls in pwm_nxp_flexio.c